hudi-common/src/test/java/org/apache/hudi/common/table/view/TestHoodieTableFileSystemView.java

```diff
@@ -1432,7 +1439,12 @@ public void testReplaceWithTimeTravel() throws IOException {
         CommitUtils.buildMetadata(Collections.emptyList(), partitionToReplaceFileIds, Option.empty(), WriteOperationType.INSERT_OVERWRITE, "", HoodieTimeline.REPLACE_COMMIT_ACTION);
     commitTimeline = metaClient.getActiveTimeline();
     HoodieInstant instant2 = new HoodieInstant(true, HoodieTimeline.REPLACE_COMMIT_ACTION, commitTime2);
-    saveAsComplete(commitTimeline, instant2, Option.of(commitMetadata.toJsonString().getBytes(StandardCharsets.UTF_8)));
+    saveAsComplete(
+        commitTimeline,
+        instant2,
+        commitMetadata instanceof HoodieReplaceCommitMetadata
+            ? serializeReplaceCommitMetadata(convertReplaceCommitMetadata((HoodieReplaceCommitMetadata) commitMetadata))
+            : serializeCommitMetadata(convertCommitMetadata(commitMetadata)));
 
```

Comment:
Fuse `serializeReplaceCommitMetadata` and `serializeCommitMetadata` into one.

Reply:
done